janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -860,10 +866,10 @@ public void clearStorage() throws BackendException {
             }
             logger.debug("Clearing storage from Solr: {}", solrClient);
             final ZkStateReader zkStateReader = ((CloudSolrClient) solrClient).getZkStateReader();
-            zkStateReader.updateClusterState();
             final ClusterState clusterState = zkStateReader.getClusterState();
-            for (final String collection : clusterState.getCollections()) {
+            for (final String collection : clusterState.getCollectionsMap().keySet()) {
                 logger.debug("Clearing collection [{}] in Solr",collection);
+                zkStateReader.forceUpdateCollection(collection);
                 final CollectionAdminRequest.Delete request = CollectionAdminRequest.deleteCollection(collection);
```

Comment:
I think you should before update cluster state with zkStateReader.forciblyRefreshAllClusterStateSlow(); (to replace zkStateReader.updateClusterState();)  because clusterState should be update before zkStateReader.getClusterState();

Reply:
@davidclement90 Thanks, good call.